spec/server/viewing/LayoutSpec.js

```diff
@@ -29,11 +34,38 @@ describe("Layout", function() {
         describe("when template has been rendered", function() {
 
             var hasDoctype = /<!doctype[^>]*>/i;
+            var preservesAttributesInHtmlTag = /<html data-type='html'>/i;
+            var preservesAttributesInBodyTag = /<body class="layout">/i;
+            var preservesAttributesInHeadTag = /<head data-type="head">/i;
```

Comment:
can you test different whitespace variations? We are using regexes now in your implementation, how do we know they are correct?